File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        lo, hi = 0, len(nums) - 1

        while lo < hi:
            m = lo + (hi - lo) // 2
            if nums[m] < nums[hi]:
                hi = m
            else:
                lo = m + 1
        return nums[lo]

File: Data Structures & Algorithms/search-insert-position/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    searchInsert(nums, target) {
        let lo = 0, hi = nums.length;
        while (lo<hi){
            let m = Math.floor(lo + (hi - lo) / 2);
            if (nums[m] === target)  return m;
            else if (nums[m] > target)  hi = m;
            else     lo = m + 1;
        }
        return lo;
    }
}
